tools/src/main/java/org/apache/kafka/tools/GroupsCommand.java

```diff
@@ -0,0 +1,298 @@
+                    .withRequiredArg()
+                    .describedAs("type")
+                    .ofType(String.class);
+
+            protocolOpt = parser.accepts("protocol", "Filter the groups based on protocol type.")
+                    .withRequiredArg()
+                    .describedAs("protocol")
+                    .ofType(String.class);
+
+            consumerOpt = parser.accepts("consumer", "Filter the groups to show all kinds of consumer groups, including classic and simple consumer groups. "
+                            + "This matches group type 'consumer', and group type 'classic' where the protocol type is 'consumer' or empty.");
+            shareOpt = parser.accepts("share", "Filter the groups to show share groups.");
+
+            options = parser.parse(args);
+
+            checkArgs();
+        }
+
+        public Boolean has(OptionSpec<?> builder) {
+            return options.has(builder);
+        }
+
+        public <A> Optional<A> valueAsOption(OptionSpec<A> option) {
+            return valueAsOption(option, Optional.empty());
+        }
+
+        public <A> Optional<A> valueAsOption(OptionSpec<A> option, Optional<A> defaultValue) {
+            if (has(option)) {
+                return Optional.of(options.valueOf(option));
+            } else {
+                return defaultValue;
+            }
+        }
+
+        public String bootstrapServer() {
+            return options.valueOf(bootstrapServerOpt);
+        }
+
+        public Properties commandConfig() throws IOException {
+            if (has(commandConfigOpt)) {
+                return Utils.loadProps(options.valueOf(commandConfigOpt));
+            } else {
+                return new Properties();
+            }
+        }
+
+        public Optional<GroupType> groupType() {
+            return valueAsOption(groupTypeOpt).map(GroupType::parse).filter(gt -> gt != GroupType.UNKNOWN);
+        }
+
+        public Optional<String> protocol() {
+            return valueAsOption(protocolOpt);
+        }
+
+        public boolean hasConsumerOption() {
+            return has(consumerOpt);
+        }
+
+        public boolean hasListOption() {
+            return has(listOpt);
+        }
+
+        public boolean hasShareOption() {
+            return has(shareOpt);
+        }
+
+        public void checkArgs() {
+            if (args.length == 0)
+                CommandLineUtils.printUsageAndExit(parser, "This tool helps to list groups of all types.");
+
+            CommandLineUtils.maybePrintHelpOrVersion(this, "This tool helps to list groups of all types.");
+
+            // should have exactly one action
+            long actions = Stream.of(listOpt).filter(options::has).count();
+            if (actions != 1)
+                CommandLineUtils.printUsageAndExit(parser, "Command must include exactly one action: --list.");
+
+            // check required args
+            if (!has(bootstrapServerOpt))
```

Comment:
Maybe this check can be replaced by `.required()`
```java
            bootstrapServerOpt = parser.accepts("bootstrap-server", "REQUIRED: The Kafka server to connect to.")
                    .withRequiredArg()
                    .describedAs("server to connect to")
                    .required()
                    .ofType(String.class);
```

Reply:
Yes, that works too, and the error message is probably nicer using `required()`.